### PR TITLE
🧹 Update mocha and remove resolutions

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description
'Extracted this change from https://github.com/gitpod-io/gitpod/pull/18734 as it's primarily chores. 

This no more:
Screenshot 2023-09-18 at 15 23 58

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 295b5d7</samp>

This pull request updates and simplifies the mocha test configuration and dependencies for several packages in the gitpod-io/gitpod repository. It also fixes a security issue with `minimatch` and removes unnecessary `mocha.opts` files and `resolutions` from the `package.json` files.

</details>

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/gitpod-db/package.json [hunk 1]
```diff
@@ -12,14 +12,28 @@
...
     "purge": "yarn clean && yarn clean:node && yarn run rimraf yarn.lock"
   },
+  "mocha": {
+    "require": [
+      "ts-node/register",
+      "reflect-metadata/Reflect",
+      "source-map-support/register"
+    ],
+    "spec": [
+      "src/**/*.spec.ts"
```
Comment (geropl) [inline components/gitpod-db/package.json:30]: This pattern does not match line 15 above. Is that relevant? :shrug:

**Comment by geropl** [comment]
Will run https://github.com/gitpod-io/gitpod/actions/workflows/lacework-inline-scanner.yml to check the docker images once we got a build.

File: components/gitpod-db/package.json (hunk 1 above)
Comment (AlexTugarev) [reply components/gitpod-db/package.json]: done

**Comment by AlexTugarev** [comment]
@geropl, on https://github.com/gitpod-io/gitpod/pull/18736#issuecomment-1723686345

No, it's just me, trying to remove unnecessary things. I've only tested with leeway and`gitpod-protocol` in the workspace and pushed to remote. But the exclude is actually useful for all dependants. The fix is building.

**Comment by geropl** [comment]
> No, it's just me, trying to remove unnecessary things. I've only tested with leeway andgitpod-protocol in the workspace and pushed to remote. But the exclude is actually useful for all dependants. The fix is building.

Noticed the `node_modules` in the path now. Works indeed! :tada:

**Comment by AlexTugarev** [comment]
Now we just™️ need to get a green build. 

> [components/server:app] Error in fetching sampling strategy: Error: connect ECONNREFUSED 0.0.0.0:5778.

🤷🏻‍♂️ this doesn't sound like a change caused by this PR.

**Comment by AlexTugarev** [comment]
Trying to understand the build problem using `leeway build components/public-api-server:app --cache=none`

**Comment by AlexTugarev** [comment]
Triggered scan in https://github.com/gitpod-io/gitpod/actions/runs/6232328813

**Review by geropl — APPROVED** [review]
:heavy_check_mark: :broom:

**Pull request merged**